### PR TITLE
support image overrides.

### PR DESCRIPTION
**Pull request by morvencao**
Signed-off-by: morvencao <lcao@redhat.com>

### CODE REVIEW

**Comment by clyang82** [comment]
Need to run `make fmt` to fix the code smell.

File: operator/pkg/config/multiclusterglobalhub_config.go [hunk 1]
```diff
@@ -43,38 +60,82 @@ func GetHoHMGHNamespacedName() types.NamespacedName {
...
-// GetImage is used to retrieve image for given component based on multiclusterglobalhub annotation
-// TODO: validate the image format is correct before return
-func GetImage(annotations map[string]string, componentName string) string {
-	if annotations != nil {
-		// dev/test only. e.g.
-		// use annotation key "hoh-hub_of_hubs_manager-image" to override multicluster-global-hub-manager image
-		componentImage, overrideImage := annotations["hoh-"+componentName+"-image"]
-		if overrideImage && componentImage != "" {
-			return componentImage
+func SetImageOverrides(mgh *operatorv1alpha1.MultiClusterGlobalHub, cm *corev1.ConfigMap) error {
```
Comment (clyang82) [inline operator/pkg/config/multiclusterglobalhub_config.go:88]: Can you add UT for this method? It can help reveal some issues.

**Review by clyang82 — CHANGES_REQUESTED** [review]

File: operator/bundle/manifests/multicluster-global-hub-operator.clusterserviceversion.yaml [hunk 2]
```diff
@@ -791,6 +791,13 @@ spec:
                 - --leader-elect
                 command:
                 - multicluster-global-hub-operator
+                env:
+                - name: OPERAND_IMAGE_MULTICLUSTER_GLOBAL_HUB_MANAGER
+                  value: quay.io/stolostron/multicluster-global-hub-manager:latest
+                - name: OPERAND_IMAGE_MULTICLUSTER_GLOBAL_HUB_AGENT
+                  value: quay.io/stolostron/multicluster-global-hub-agent:latest
+                - name: OPERAND_IMAGE_MULTICLUSTER_GLOBAL_HUB_OPERATOR
```
[the remark this answers is not in the document]
Comment (morvencao) [reply operator/bundle/manifests/multicluster-global-hub-operator.clusterserviceversion.yaml:799]: yes, for console setup and clean.

**Comment by morvencao** [comment]
Added UT for image overrides, the only issue is that the some lines of map literal can not be splitted.

File: operator/pkg/config/multiclusterglobalhub_config.go (hunk 1 above)
Comment (morvencao) [reply operator/pkg/config/multiclusterglobalhub_config.go:88]: Added UT.

**Pull request merged**